cmd/setup-etcd-environment/run.go

```diff
@@ -150,11 +156,16 @@ func reverseLookupSelf(service, proto, name, self string) (string, error) {
 	return selfTarget, nil
 }
 
-func writeEnvironmentFile(m map[string]string, w io.Writer) error {
+func writeEnvironmentFile(m map[string]string, w io.Writer, export bool) error {
 	var buffer bytes.Buffer
 	for k, v := range m {
 		buffer.WriteString(fmt.Sprintf("ETCD_%s=%s\n", k, v))
 	}
+	if export == true {
+		s := buffer.String()
+		buffer.Reset()
+		buffer.WriteString(fmt.Sprintf("set -a\n%sset +a\n", s))
```

Comment:
the reason why we want to export certain is to differentiate the access between the env variables available to the script calls `exec etcd` and the `etcd` process itself right?

would it be simpler, if all envs from setup-etcd-environment should be available to both ie set all as export.

Reply:
> would it be simpler, if all envs from setup-etcd-environment should be available to both ie set all as export.

it would be easier but as we prefix `ETCD_` on all env they get picked up by the binary and print a bit of cruft during startup. Stuff like

2019-07-11 20:36:38.756064 W | pkg/flags: unrecognized environment variable ETCD_IPV4_ADDRESS=192.168.0.12

While this is mostly cosmetic I fear it could cause confusion.

Reply:
+1

Reply:
+1

i think this reasoning would be useful in the commit message or the PR description for future reference.

Reply:
good idea, updated